tests/system/action/list_of_speakers/test_re_add_last.py

```diff
@@ -112,3 +113,70 @@ def test_last_speaker_also_in_waiting_list(self) -> None:
         self.assertTrue(
             "User 42 is already on the list of speakers." in response.json["message"]
         )
+
+    def test_r_add_last_no_permissions(self) -> None:
+        self.base_permission_test(
+            {
+                "user/42": {"username": "test_username42", "speaker_$222_ids": [222]},
+                "user/43": {"username": "test_username43", "speaker_$222_ids": [223]},
+                "user/44": {"username": "test_username43", "speaker_$222_ids": [224]},
+                "list_of_speakers/111": {
+                    "closed": False,
+                    "meeting_id": 1,
+                    "speaker_ids": [222, 223, 224],
+                },
+                "speaker/222": {
+                    "list_of_speakers_id": 111,
+                    "user_id": 42,
+                    "begin_time": 1000,
+                    "end_time": 2000,
+                },
+                "speaker/223": {
+                    "list_of_speakers_id": 111,
+                    "user_id": 43,
+                    "begin_time": 3000,
+                    "end_time": 4000,
+                },
+                "speaker/224": {
+                    "list_of_speakers_id": 111,
+                    "user_id": 44,
+                    "begin_time": 5000,
+                },
+            },
+            "list_of_speakers.re_add_last",
+            {"id": 111},
+        )
+
+    def test_r_add_last_permissions(self) -> None:
+        self.base_permission_test(
+            {
+                "user/42": {"username": "test_username42", "speaker_$222_ids": [222]},
+                "user/43": {"username": "test_username43", "speaker_$222_ids": [223]},
+                "user/44": {"username": "test_username43", "speaker_$222_ids": [224]},
+                "list_of_speakers/111": {
+                    "closed": False,
+                    "meeting_id": 1,
+                    "speaker_ids": [222, 223, 224],
+                },
+                "speaker/222": {
+                    "list_of_speakers_id": 111,
+                    "user_id": 42,
+                    "begin_time": 1000,
+                    "end_time": 2000,
+                },
+                "speaker/223": {
+                    "list_of_speakers_id": 111,
+                    "user_id": 43,
+                    "begin_time": 3000,
+                    "end_time": 4000,
+                },
+                "speaker/224": {
+                    "list_of_speakers_id": 111,
+                    "user_id": 44,
+                    "begin_time": 5000,
+                },
+            },
```

Comment:
if this is the same object as above, consider saving it in a variable instead of explicitly typing it twice

Reply:
Ah, move thís into permission_test_model in setUp.